### PR TITLE
[parse] Fix for const enum throwing isolatedModules error

### PR DESCRIPTION
**Pull request by llkats**
When attempting to use `Parse.ErrorCode` in a project compiled by Babel, I'm running into the error: 
`Type error: Ambient const enums are not allowed when the '--isolatedModules' flag is provided.`

Apparently `const enums` are [not recommended in .d.ts files](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/23002#issuecomment-359051201). 

I found abandoned PR [#31248](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/31248) and implemented the feedback suggested by reviewer @agoldis. I'm fairly new to TypeScript myself, so please do let me know if there are any further changes to be made.

---

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/31248
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/parse/index.d.ts [hunk 1]
```diff
@@ -19,6 +19,64 @@
 
 /// <reference types="node" />
 
+declare enum ErrorCode {
+    OTHER_CAUSE = -1,
```
Comment (uniqueiniquity) [inline types/parse/index.d.ts]: If you want to be able to import these from a consuming file, they need to be within the `Parse` namespace.

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

File: types/parse/index.d.ts (hunk 1 above)
Comment (llkats) [reply types/parse/index.d.ts]: @uniqueiniquity they are exported as static variables on the `Parse.Error` class in the changes below, [starting at line 968](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38472/files#diff-aec51b2d87715b12fe03e0f09983ea8fR968). I followed the advice from the previous abandoned PR in [this comment](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/31248#pullrequestreview-183468400). The reasoning behind not declaring the enum inside the class is that the entity `Parse.ErrorCode` doesn't exist on Parse object.

The import appears to work when I test this file in my local `node_modules` folder. Is there something more that I'm not understanding? Thanks for your help!

Comment (uniqueiniquity) [reply types/parse/index.d.ts]: Makes sense to me, I just wanted to check. Thanks for the explanation!

**Review by uniqueiniquity — APPROVED** [review]

**Comment by uniqueiniquity** [comment]
Congratulations on your first DefinitelyTyped contribution!
Thank you for being a part of the community!

**Comment by llkats** [comment]
@uniqueiniquity thank you!! 🎉

**Pull request merged**